.circleci/config.yml

```diff
@@ -223,10 +225,14 @@ jobs:
           name: run c++ tests with meson
           command: |
             . env/bin/activate
-            meson setup _build -Dbuildtype=debug -Dwerror=true
+            meson setup _build -Dbuildtype=debug -Dwerror=true $(if [ "<< parameters.cpp-standard >>" != "default" ]; then echo "-Dcpp_std=<< parameters.cpp-standard >>"; fi)
```

Comment:
If we want to do conditional, then something like also works.
```
run:
    command: ...
    when: { equal: [<< parameters.cpp-standard >>, "default" ]}
run:
    command: ...
    when: {not : {equal : [<< parameters.cpp-standard >>, "default" ] }}
```

That said, IMO we should just explicitly sweep compiler versions, see other comment